src/tunnel.cc

```diff
@@ -1357,8 +1384,10 @@ TunnelStateData::startConnecting()
 
     assert(!destinations->empty());
     assert(!transporting());
+    forgetError();
+    al->http.code = Http::scNone;
```

Comment:
I think we need this cleanup for both startConnecting() callers: retryOrBail() and noteDestination().